# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../rt2500/2.6.14.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ur rt2500-1.1.0_orig/Module/rtmp_info.c rt2500-1.1.0/Module/rtmp_info.c
--- rt2500-1.1.0_orig/Module/rtmp_info.c	2005-02-22 11:13:39.000000000 +0100
+++ rt2500-1.1.0/Module/rtmp_info.c	2005-11-04 02:08:54.000000000 +0100
@@ -1970,7 +1970,7 @@
             break;
         case SIOCGIWPRIV:
             if (wrq->u.data.pointer) {
-                if ( verify_area(VERIFY_WRITE, wrq->u.data.pointer, sizeof(privtab)) )
+                if ( !access_ok(VERIFY_WRITE, wrq->u.data.pointer, sizeof(privtab)) )
                     break;
                 wrq->u.data.length = sizeof(privtab) / sizeof(privtab[0]);
                 if (copy_to_user(wrq->u.data.pointer, privtab, sizeof(privtab)))
@@ -1983,7 +1983,7 @@
                 char *this_char;
                 char *value;
 
-                if( verify_area(VERIFY_READ, wrq->u.data.pointer, wrq->u.data.length) )
+                if( !access_ok(VERIFY_READ, wrq->u.data.pointer, wrq->u.data.length) )
                     break;
 
                 while ((this_char = strsep((char**)&wrq->u.data.pointer, ",")) != NULL) 
